### PR TITLE
Feature/je 339 external invoices extra fields

### PR DESCRIPTION
**Pull request by tuj**
https://jira.itkdev.dk/browse/JE-339

* Changed project billing to look for 'status=lukket' instead of 'status=done'.
* Added new fields to external invoices.
* Fixed invoice preview table styling.

### CODE REVIEW

File: bundles/Billing/Resources/assets/pages/Invoice.js [hunk 1]
```diff
@@ -81,7 +84,9 @@ class Invoice extends Component {
                     formPaidByAccount: response.paidByAccount ? response.paidByAccount : '',
                     formDefaultPayToAccount: response.defaultPayToAccount ? response.defaultPayToAccount : '',
                     formDefaultMaterialNumber: response.defaultMaterialNumber ? response.defaultMaterialNumber : '',
-                    formAccount: response.accountId ? response.accountId : ''
+                    formAccount: response.accountId ? response.accountId : '',
+                    formPeriodFrom: response.periodFrom ? parseInt(response.periodFrom) * 1000 : '',
+                    formPeriodTo: response.periodTo ? parseInt(response.periodTo) * 1000 : ''
```
Comment (rimi-itk) [inline bundles/Billing/Resources/assets/pages/Invoice.js]: It seems weird that `formPeriodFrom` and `formPeriodTo` can be either an `int` or a `string` – but it works!

File: bundles/ProjectBilling/Service/ProjectBillingService.php [hunk 2]
```diff
@@ -305,6 +308,19 @@ public function exportTasksToSpreadsheet(array $entries)
...
+
+        // Move ahead if the day is a saturday or sunday to ensure it is a bank day.
+        // TODO: Handle holidays.
+        $weekday = $todayPlus30days->format('N');
+        if ('6' === $weekday) {
+            $todayPlus30days->add(new \DateInterval('P2D'));
+        } elseif ('7' === $weekday) {
+            $todayPlus30days->add(new \DateInterval('P1D'));
+        }
+
```
Comment (rimi-itk) [inline bundles/ProjectBilling/Service/ProjectBillingService.php:323]: You could use `$todayPlus30days = $today->modify('30 days')->modify('monday');`.

File: bundles/ProjectBilling/Service/ProjectBillingService.php [hunk 3]
```diff
@@ -338,6 +354,22 @@ public function exportTasksToSpreadsheet(array $entries)
...
+                // 39. Periode fra
+                $sheet->setCellValueByColumnAndRow(39, $row, $header->periodFrom);
+                // 40. Periode til
+                $sheet->setCellValueByColumnAndRow(40, $row, $header->periodTo);
+                // 46. Fordringstype oprettelse/valg : KOCIVIL
+                $sheet->setCellValueByColumnAndRow(46, $row, 'KOCIVIL');
+                // 49. Forfaldsdato: dagsdato
+                $sheet->setCellValueByColumnAndRow(49, $row, $todayString);
+                // 50. Henstand til: dagsdato + 30 dage. NB det må ikke være før faktura forfald. Skal være en bank dag.
+                $sheet->setCellValueByColumnAndRow(50, $row, $todayPlus30days->format('d.m.Y'));
```
Comment (rimi-itk) [inline bundles/ProjectBilling/Service/ProjectBillingService.php]: For consistency you could define `$todayPlus30daysString` where you define `$todayString` and use it here.

File: bundles/Billing/Service/BillingService.php [hunk 4]
```diff
@@ -188,6 +188,8 @@ private function getInvoiceArray(Invoice $invoice, bool $withAccount = false)
             'created_by' => $invoice->getCreatedBy(),
             'defaultPayToAccount' => $invoice->getDefaultPayToAccount(),
             'defaultMaterialNumber' => $invoice->getDefaultMaterialNumber(),
+            'periodFrom' => $invoice->getPeriodFrom() ? $invoice->getPeriodFrom()->format('U') : null,
+            'periodTo' => $invoice->getPeriodTo() ? $invoice->getPeriodTo()->format('U') : null,
```
Comment (rimi-itk) [inline bundles/Billing/Service/BillingService.php:192]: For consistency with `exportedDate` you could use `->format('c')` here and use `new Date(…).getTime()` (or `moment`) to convert the date value in https://github.com/itk-dev/jira_economics/pull/109/files#diff-feb28d681f25eea707a612b967b33048192c4c722fcc8f2512014cd2af263b38R88-R89.

**Review by rimi-itk — APPROVED** [review]
Looks good. Some suggestions on handling of dates added.

File: bundles/ProjectBilling/Service/ProjectBillingService.php (hunk 2 above)
Comment (tuj) [reply bundles/ProjectBilling/Service/ProjectBillingService.php:323]: Nice :)

**Pull request merged**